semantic-python/test/Test.hs

```diff
@@ -76,13 +78,23 @@ assertJQExpressionSucceeds directive tree core = do
 fixtureTestTreeForFile :: HasCallStack => Path.RelFile -> Tasty.TestTree
 fixtureTestTreeForFile fp = HUnit.testCaseSteps (Path.toString fp) $ \step -> withFrozenCallStack $ do
   let fullPath = Path.relDir "semantic-python/test/fixtures" </> fp
+      perish s = liftIO (HUnit.assertFailure ("Directive parsing error: " <> s))
 
-  fileContents <- ByteString.readFile (Path.toString fullPath)
-  directives <- case Directive.parseDirectives fileContents of
-    Right dir -> pure dir
-    Left err  -> HUnit.assertFailure ("Directive parsing error: " <> err)
+  -- Slurp the input file, taking lines from the beginning until we
+  -- encounter a line that doesn't have a '#'. For each line, parse
+  -- a directive out of it, failing if the directive can't be parsed.
+  directives <-
+    runResourceT
+    . Stream.toList_
+    . Stream.mapM (either perish pure . Directive.parseDirective)
+    . Stream.takeWhile ((== '#') . ByteString.head)
+    . Stream.mapped ByteStream.toStrict
+    . ByteStream.denull
+    . ByteStream.lines
+    . ByteStream.readFile @(ResourceT IO)
+    $ Path.toString fullPath
```

Comment:
Once we have comment support in `tree-sitter-python` ASTs (https://github.com/tree-sitter/haskell-tree-sitter/issues/99) we can do this in an easier fashion (i.e. slurp the top-level comment nodes from the current module), but this will do for now.